modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/UserEvent.scala

```diff
@@ -17,66 +17,71 @@ import scalaz.syntax.std.option._
 /**
   * Events generated by the user.
   */
-sealed trait UserEvent {
+sealed trait UserEvent[+D] {
```

Comment:
Because if you do that, then you have `final case class Foo[D] extends UserEvent`. If you then match on a value of type `UserEvent`, you have to consider all the possible `Foo` for every possible `D` (which does not have a restriction, so it can be anything)

Reply:
Is the difference between only `Foo[D]` can be of type `UserEvent[D]`, and any `Foo[A]` can be of type `UserEvent`.